src/libexpr/eval.cc

```diff
@@ -1621,15 +1621,16 @@ void EvalState::callFunction(Value & fun, size_t nrArgs, Value * * args, Value &
                 return;
             } else {
                 /* We have all the arguments, so call the primop. */
-                auto name = vCur.primOp->name;
+                auto fn = vCur.primOp;
 
                 nrPrimOpCalls++;
-                if (countCalls) primOpCalls[name]++;
+                if (countCalls && !fn->hideInDiagnostics) primOpCalls[fn->name]++;
```

Comment:
Technically a primop call, so that should be ok.

```suggestion
                // This will count black holes, but that's ok, because unrecoverable errors are rare.
                if (countCalls) primOpCalls[fn->name]++;
```